mailbox/api/src/main/java/org/apache/james/mailbox/model/ContentType.java

```diff
@@ -39,6 +192,30 @@ public ContentType(String value) {
         this.value = value;
     }
 
+    public ContentTypeField asMime4J() {
+        return Fields.contentType(value);
+    }
+
+    public MimeType mimeType() {
+        ContentTypeField contentTypeField = asMime4J();
+        return MimeType.of(
+            MediaType.of(contentTypeField.getMediaType()),
+            SubType.of(contentTypeField.getSubType()));
+    }
+
+    public MediaType mediaType() {
+        return MediaType.of(asMime4J().getMediaType());
+    }
+
+    public SubType subType() {
+        return SubType.of(asMime4J().getSubType());
+    }
+
+    public Optional<Charset> charset() {
+        return Optional.ofNullable(asMime4J().getCharset())
```

Comment:
I think it would be nicer to have the mediatype and charset as field.
And use `asMime4J` only in a factory method ?

Reply:
- I don't want to alter the raw unparsed value (as it may break things in tests and make adoption harder)
 - There might be other parameters: name for instance.
 - We currently allow invalid value as part of header fields. Agressively catching them might lead to some issues.

IMO the first step is to get the API correct, then moving where we do the parsing is easy.

One step at a time?